### PR TITLE
Adding How-to delete data

### LINKED ISSUE #3999: Docs: How to delete data

### PR DESCRIPTION
**Pull request by giovannipizzi**
Fixes #3999

### CODE REVIEW

File: docs/source/howto/data.rst [hunk 1]
```diff
@@ -276,7 +276,52 @@ Sharing data
 Deleting data
 =============
 
-`#3999`_
+By default, every time you run or submit a new calculation or workflow, AiiDA will create for you new nodes in the database, and will never replace or delete data.
```
Comment (csadorf) [inline docs/source/howto/data.rst:279]:
```suggestion
By default, every time you run or submit a new calculation, AiiDA will create for you new nodes in the database, and will never replace or delete data.
```

File: docs/source/howto/data.rst [hunk 2]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+There are cases, however, when it might be useful to delete nodes that are not useful anymore, for instance test runs or incorrect/wrong data and calculations.
+In this case, AiiDA provides a ``verdi node delete`` command to remove the nodes from the provenance graph.
```
Comment (csadorf) [inline docs/source/howto/data.rst:281]:
```suggestion
For this case, AiiDA provides the ``verdi node delete`` command to remove the nodes from the provenance graph.
```

File: docs/source/howto/data.rst [hunk 3]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+
+.. note::
```
Comment (csadorf) [inline docs/source/howto/data.rst:283]:
```suggestion
.. caution::
```

File: docs/source/howto/data.rst [hunk 4]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+   The ``verdi node delete`` command should be used with great care.
+
+   First, once the data is deleted, there is no way to recover it (unless you made a backup).
```
Comment (csadorf) [inline docs/source/howto/data.rst:286]:
```suggestion
   Once the data is deleted, there is no way to recover it (unless you made a backup).
```

File: docs/source/howto/data.rst [hunk 5]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+
+   Even more critically, even if you ask to delete only one node, ``verdi node delete`` will typically delete a number of additional linked nodes, in order to preserve a consistent state of the provenance graph.
+   For instance, if you delete an input of a calculation, AiiDA will delete also the calculation itself (as otherwise you would be effectively changing the inputs to that calculation in the provenance graph).
+   The full set of consistency rules are explained in detail :ref:`here <topics:provenance:consistency>`.
+
+   Therefore: always check the output of ``verdi node delete`` to make sure that it is not deleting more than you expect.
+   You can also use the ``--dry-run`` flag of ``verdi node delete`` to see what the command would do without performing any actual operation.
+
+In addition, there are a number of additional rules that are not mandatory to ensure consistency, but can be toggled by the user.
```
Comment (csadorf) [inline docs/source/howto/data.rst:295]: Is this referring to the text in the "note" block? Usually one should be able to assume that the flow of the text is not interrupted if one omits reading the extra blocks.

File: docs/source/howto/data.rst [hunk 6]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+Deleting mutable data
+---------------------
+A subset of data in AiiDA is mutable also after storing a node, and is used as a convenience for the user to tag/group/comment on data.
+This data can be safely deleted at any time.
+This includes, notably:
+
+* *Node extras*: These can be deleted using :py:meth:`~aiida.orm.nodes.node.Node.delete_extra` and :py:meth:`~aiida.orm.nodes.node.Node.delete_extra_many`.
+* *Node comments*: These can be removed using :py:meth:`~aiida.orm.nodes.node.Node.remove_comment`.
+* *Groups*: These can be deleted using :py:meth:`Group.objects.delete() <aiida.orm.groups.Group.Collection.delete>`.
+  Note that only the group is deleted, while the nodes contained in it are not deleted.
```
Comment (csadorf) [inline docs/source/howto/data.rst:314]:
```suggestion
  This command will only delete the group, not the nodes contained in the group.
```

File: docs/source/howto/data.rst [hunk 7]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+
+Completely deleting an AiiDA profile
+------------------------------------
+If you don't want to selectively delete some nodes, but you want to delete a whole AiiDA profile altogether, use the ``verdi profile delete`` command.
```
Comment (csadorf) [inline docs/source/howto/data.rst:318]:
```suggestion
If you don't want to selectively delete some nodes, but instead want to delete a whole AiiDA profile altogether, use the ``verdi profile delete`` command.
```

File: docs/source/howto/data.rst [hunk 8]
```diff
@@ -276,7 +276,52 @@ Sharing data
...
+This command will delete both the file repository and the database.
+
+.. note::
+
+  This operation cannot be undone.
```
Comment (csadorf) [inline docs/source/howto/data.rst:323]: I'm not sure that this warning is *really* necessary, but if we keep then it should maybe be a bit more explicit.
```suggestion
.. danger::

  It is not possible to restore a deleted profile unless it was previously backed up!
```

File: docs/source/howto/data.rst (hunk 5 above)
Comment (giovannipizzi) [reply docs/source/howto/data.rst:293]: Indeed - I also had the same doubt. I now left only a minimal note/caution block, and moved the content outside. The other option was to move everything in the block but it seemed less elegant.

**Comment by giovannipizzi** [comment]
@csadorf should be ready to be reviewed again

**Pull request merged**